libnetwork/networkdb/networkdb.go

```diff
@@ -337,15 +337,17 @@ func (nDB *NetworkDB) Peers(nid string) []PeerInfo {
 func (nDB *NetworkDB) GetEntry(tname, nid, key string) ([]byte, error) {
 	nDB.RLock()
 	defer nDB.RUnlock()
-	entry, err := nDB.getEntry(tname, nid, key)
+	v, err := nDB.getEntry(tname, nid, key)
 	if err != nil {
 		return nil, err
 	}
-	if entry != nil && entry.deleting {
+	if v != nil && v.deleting {
 		return nil, types.NotFoundErrorf("entry in table %s network id %s and key %s deleted and pending garbage collection", tname, nid, key)
 	}
-
-	return entry.value, nil
+	if v == nil {
+		return nil, nil
+	}
```

Comment:
Why is this check needed with v2 but not with v1?

Reply:
Good question; maybe it was needed as well; there's a nil-check two lines up, and my IDE was flagging it as "could be nil"; honestly not sure it there's such a chance

Reply:
There is at least one other place which assumes that `(*NetworkDB).getEntry() error = nil` implies that the returned `*entry` is not nil.
https://github.com/moby/moby/blob/c2879978022db54ef21ac93feeabce0cbd75f63d/libnetwork/networkdb/delegate.go#L171-L179
I am pretty sure that it is an invariant. An incomplete audit of the NetworkDB code suggests that to be the case.